pippy/PipelineDriver.py

```diff
@@ -1184,7 +1184,7 @@ def run_until(self, predicate: Callable[[torch.fx.Node], bool]):
             if predicate(node):
                 return node
 
-            return self.run_one(node)
+            self.run_one(node)
```

Comment:
Nit, also add `-> Optional[Node]` return annotation to the `run_until` signature. Maybe also add an `assert False, 'No node returned True for predicate'` after the while loop here

Reply:
Actually no call site of `run_until` requires any return -- shall we make this function just return nothing?

Reply:
It's theoretically useful for `run_until` to report when it stopped, so I would elect to keep the return

Reply:
Got it, will add `-> Optional[Node]`.
For the `assert False` call, wouldn't it make things fail here?
https://github.com/pytorch/PiPPy/blob/main/pippy/PipelineDriver.py#L1038

Reply:
I will probably just add `return None` after the while loop, meaning no more node to run :)

Reply:
Sure, I think the semantics here were not actually that clear since the function didn't actually work properly, but it's a fine decision to make it so that the FN runs until the end regardless of whether the predicate is True